version: '3'

services:

  db:
    image: postgres:alpine
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      PGDATA: /var/lib/postgresql/data/pgdata
    volumes:
      - .pg_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build: .
    tty: true
    command: >
      sh -c "python manage.py makemigrations &&
         python manage.py migrate &&
         python manage.py load_locations &&
         python manage.py load_cars &&
         python manage.py runserver 0.0.0.0:8000"
    ports:
      - '8000:8000'
    volumes:
      - .:/app
    depends_on:
      - redis
      - celery
      - celery_beat

  redis:
    image: redis:alpine
    ports:
      - '6379:6379'
    depends_on:
      db:
        condition: service_healthy

  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO
    volumes:
      - .:/app
    depends_on:
      - redis

  celery_beat:
    build: .
    tty: true
    command: celery -A config beat -l INFO
    volumes:
      - .:/app
    depends_on:
      - redis

volumes:
  pg_data: